tests/dotnet/UnitTests/AssetsTest.cs

```diff
@@ -0,0 +1,221 @@
+	[TestFixture (ApplePlatform.MacCatalyst, "maccatalyst-x64", "macosx", true)]
+	[TestFixture (ApplePlatform.MacCatalyst, "maccatalyst-arm64;maccatalyst-x64", "macosx", true)]
+	[TestFixture (ApplePlatform.MacOSX, "osx-x64", "macosx", true)]
+	[TestFixture (ApplePlatform.MacOSX, "osx-arm64;osx-x64", "macosx", true)] // https://github.com/xamarin/xamarin-macios/issues/12410
+	// Build, add the XCAssets, then build again
+	[TestFixture (ApplePlatform.iOS, "iossimulator-x64", "iphonesimulator", false)]
+	[TestFixture (ApplePlatform.iOS, "ios-arm64;ios-arm", "iphoneos", false)]
+	[TestFixture (ApplePlatform.TVOS, "tvossimulator-x64", "appletvsimulator", false)]
+	[TestFixture (ApplePlatform.MacCatalyst, "maccatalyst-x64", "macosx", false)]
+	[TestFixture (ApplePlatform.MacCatalyst, "maccatalyst-arm64;maccatalyst-x64", "macosx", false)]
+	[TestFixture (ApplePlatform.MacOSX, "osx-x64", "macosx", false)]
+	[TestFixture (ApplePlatform.MacOSX, "osx-arm64;osx-x64", "macosx", false)] // https://github.com/xamarin/xamarin-macios/issues/12410
+	public class AssetsTest : TestBaseClass {
+
+		readonly ApplePlatform platform;
+		readonly string runtimeIdentifiers;
+		readonly string sdkVersion;
+		readonly bool isStartingWithAssets;
+		string projectPath = string.Empty;
+		string appPath = string.Empty;
+
+		public AssetsTest (ApplePlatform platform, string runtimeIdentifiers, string sdkVersion, bool isStartingWithAssets)
+		{
+			this.platform = platform;
+			this.runtimeIdentifiers = runtimeIdentifiers;
+			this.sdkVersion = sdkVersion;
+			this.isStartingWithAssets = isStartingWithAssets;
+		}
+
+		[SetUp]
+		public void Init ()
+		{
+			var project = "AppWithXCAssets";
+			Configuration.IgnoreIfIgnoredPlatform (platform);
+			projectPath = GetProjectPath (project, runtimeIdentifiers: runtimeIdentifiers, platform: platform, out appPath);
+			ConfigureAssets ();
+		}
+
+		[TearDown]
+		public void Cleanup ()
+		{
+			DeleteAssets ();
+		}
+
+		[Test]
+		public void TestXCAssets ()
+		{
+			var appExecutable = GetNativeExecutable (platform, appPath);
+			DotNetProjectTest.ExecuteWithMagicWordAndAssert (platform, runtimeIdentifiers, appExecutable);
+
+			var resourcesDirectory = GetResourcesDirectory (platform, appPath);
+
+			var assetsCar = Path.Combine (resourcesDirectory, "Assets.car");
+			Assert.That (assetsCar, Does.Exist, "Assets.car");
+
+			var doc = ProcessAssets (assetsCar, GetFullSdkVersion ());
+			Assert.IsNotNull (doc, "There was an issue processing the asset binary.");
+
+			var foundAssets = FindAssets (doc);
+
+			// Seems the 2 vectors are not being consumed in MacCatalyst but they still appear in the image Datasets
+			var TotalUniqueAssets = platform == ApplePlatform.MacCatalyst ? 14 : 16;
+
+			Assert.AreEqual (TotalUniqueAssets, foundAssets.Count, "Wrong number of assets found");
+			Assert.That (foundAssets, Has.No.Member ("Data.DS_StoreDataTest"), "DS_Store files should not be included.");
```

Comment:
I think it would be better to assert that the entire set of assets is as expected.

So list all the expected assets (in a hashset for instance), and then assert that the expected assets and the found assets all have the same paths.